packages/wonder-blocks-form/components/choice-internal.js

```diff
@@ -0,0 +1,132 @@
+import getClickableBehavior from "../../wonder-blocks-core/util/get-clickable-behavior";
+import CheckboxCore from "./checkbox-core.js";
+import RadioCore from "./radio-core.js";
+
+type Props = {|
+    /** Label for the field. */ label: string,
+    /** Optional description for the field. */ description?: string,
+    /** Should be distinct for each item in the group. */ value?: string,
+    /** Whether this choice is checked. */ checked: boolean,
+    /** Whether this choice option is disabled. */ disabled: boolean,
+    /** Whether this choice is in error mode. */ error: boolean,
+    /** Autopopulated by parent's groupName prop if in a group. */ groupName?: string,
+    /** Used for accessibility purposes, where the label id should match the
+     * input id. */ id?: string,
+    /** Returns the new checked state of the component. */ onChange: (
+        newCheckedState: boolean,
+    ) => void,
+    /**  Optional additional styling. */ style?: any,
+    /** Optional id for testing purposes. */ testId?: string,
+    /** Takes either "radio" or "checkbox" value. */ variant:
+        | "radio"
+        | "checkbox",
+|};
+
+export default class ChoiceInternal extends React.Component<Props> {
+    static defaultProps = {
+        checked: false,
+        disabled: false,
+        error: false,
+    };
+
+    getChoiceCoreComponent() {
+        if (this.props.variant === "radio") {
+            return RadioCore;
+        } else {
+            return CheckboxCore;
+        }
+    }
+    render() {
+        const {
+            label,
+            description,
+            onChange,
+            style,
+            variant,
+            // we don't need this to go into coreProps
+            // eslint-disable-next-line no-unused-vars
+            value,
+            ...coreProps
+        } = this.props;
+        const ClickableBehavior = getClickableBehavior();
+
+        const Choice = this.getChoiceCoreComponent();
+
+        return (
+            <View style={style}>
+                <ClickableBehavior
+                    disabled={coreProps.disabled}
+                    onClick={(e) => {
+                        // Radio buttons cannot be unchecked and do not change
+                        // if clicked on when checked
+                        if (variant === "radio" && coreProps.checked) {
+                            return;
+                        }
+                        onChange(!coreProps.checked);
+                    }}
+                >
+                    {(state, handlers) => {
+                        return (
+                            <View
+                                style={[
+                                    styles.wrapper,
+                                    state.hovered && styles.hovered,
+                                    style,
+                                ]}
+                                {...handlers}
+                                tabIndex={-1}
```

Comment:
Is this tabIndex needed? Would you ever be able to tab to this view? If the tabIndex is coming from `handlers` it would be good to have a comment explaining as such.